tools/cleanup_duplicate_host_guest_devices.rb

```diff
@@ -45,6 +46,20 @@
 
 return if opts[:dry_run]
 
-GuestDevice.destroy(guest_devices_to_delete) unless opts[:dry_run]
+dependents = GuestDevice.reflect_on_all_associations(:has_one) + GuestDevice.reflect_on_all_associations(:has_many)
+dependent_destroys = dependents.select { |assoc| assoc.options[:dependent] == :destroy }
+
+total_slices = (guest_devices_to_delete.count / opts[:page_size].to_f).ceil
+guest_devices_to_delete.each_slice(opts[:page_size]).with_index do |slice, index|
+  puts "Destroying slice #{index + 1} of #{total_slices}..."
+  GuestDevice.delete(slice)
+
+  dependent_destroys.each do |dependent_assoc|
+    foreign_key = dependent_assoc.options[:foreign_key] || "guest_device_id"
```

Comment:
pretty sure this will work for us:

```suggestion
    foreign_key = dependent_assoc.join_keys.foreign_key
```

Reply:
Had to be `join_keys.key` but yeah that worked :+1: